### ISSUE TITLE
[Feature-request] making the search more "fuzzy"

### ISSUE THREAD

**Issue by revuwa** [Iamlooker/Droid-ify#87]
If I search (e. g. unter "Available" apps) for "app list", I got other results as if I search for "list app".

For example:

- "app list" lists apps like: [My **App List**](https://f-droid.org/en/packages/com.projectsexception.myapplist.open/), and
- "list app" lists apps like: [**List** My **App**s](https://f-droid.org/en/packages/de.onyxbits.listmyapps/)

but not vice versa, means I can't find [List My Apps](https://f-droid.org/en/packages/de.onyxbits.listmyapps/) with the search string "app list".

I don't know if this is intentional?
If not, I would prefer to find all apps with "app" & "list" in their names, by searching with both of the search strings above.

One step cooler would be a real fuzzy search, or what do you think?

**Comment by endim8** [comment]
Definitely agree with this, a good start would be to search the short app description as well as the appname

**Comment by vertigo220** [comment]
Could do something similar to G-Droid's search harder.

**Comment by vertigo220** [comment]
I'm finding the search to be pretty questionable. For example, I searched for "home hue" (without quotes) and there were no results, despite the fact the app I was looking for is named "Home App | For Philips Hue, Arduino & more" and so the search definitely should have found it. I wonder how many apps I've not found when doing various searches.

**Comment by jcperil** [comment]
I have a similar experience. I searched for the keywords "share location" and "share your location" and I received mixed results.

<p float="left">
<img src="https://github.com/Iamlooker/Droid-ify/assets/89805719/89a1ec38-ec16-478f-b689-931e1e635aaa" width="160px" />
<img src="https://github.com/Iamlooker/Droid-ify/assets/89805719/d590520c-8f70-40ff-923f-c692ece309c8" width="160px" />
</p>

---

Can something like [fzf](https://github.com/junegunn/fzf) be used for Android?

**Comment by gallegonovato** [comment]
Good afternoon @Iamlooker

What I do notice, is that for example an application shows it twice.

One with fdroid source and the other Izzy.

I do not know if they could be unified?

But for example with Magisk all the sources together.

Best regards

**Comment by at-pyrix** [comment]
Any updates on this?

**Comment by MikeSesketcharba** [comment]
I also want this

**Comment by e-t-l** [comment]
> The strictness of the search makes it strange to use. Even a trailing space will change the results list. Seems like maybe it does search descriptions but only for precise matches.

Yes. The problem is that it is searching for the entire string, instead of a set of keywords like most search engines would do by default. Usually, an exact string search should be enclosed in quotation marks.

> Can something like fzf be used for Android?

From  a quick look at that repo, I don't *think* it will work; that's a whole command-line program for desktop OSes. We would need something that's like a Java or Kotlin library (bundle of code that can be included in a project).

> Searching `everydoor` returns no results,
> Searching `every door` returns correctly.

I don't know that that's the same issue. If you're looking for an app that contains the word "every" and "door" in the title or description, **then you need to put a space between those, to make them separate keywords**. If you expect it to find that app when you search the single keyword "everydoor", then that means it would be searxh in every *substring* od every keyword: It would also return any result that contains "e" "ver" "do" "or" etc. That's not very useful!

**Comment by opk12** [comment]
When I type `a b c` I mean `*a* AND *b* AND *c*`, that is, split at space + full-text search the AND of the pieces.

Other projects use SQLite's [FTS extension](https://sqlite.org/fts5.html)...
* f-droid.org's [search webpage](https://gitlab.com/fdroid/fdroid-website-search/-/work_items) (not to be confused with the fdroid-website repo)
  * you can see the queries in the file [views.py](https://gitlab.com/fdroid/fdroid-website-search/-/blob/master/fdroid_website_search/views.py)
* [Another Notes App](https://f-droid.org/en/packages/com.maltaisn.notes.sync/)

... but expect bugreports for FTS's own problems and counter-intuitive results
* it's whole-word, so [Game Pad does not find GamePad](https://gitlab.com/fdroid/fdroid-website-search/-/work_items/71) - [stor does not find store](https://gitlab.com/Benjamin_Loison/fdroid-website-search/-/issues/2) - [th does not find GitHub](https://github.com/maltaisn/another-notes-app/issues/154)
* for Unicode you're limited to what your version of FTS supports, see [here](https://github.com/maltaisn/another-notes-app/issues/155) (example of what may be affected: DAVx⁵)
